### PR TITLE
Add `sub_type` search option to rules

### PR DESCRIPTION
**Pull request by Berthe01**
Add search option to be able to filter rules by subtype using RestAPI.

I just don't know hopw to choose the ID of this searchOption :/

### CODE REVIEW

**Comment by stonebuzz** [comment]
To determine ID you can you this

```shell
php tools/getsearchoptions.php --type=Rule
Array
(
    [1] => N/A / Nom
    [2] => N/A / ID
    [3] => N/A / Classement
    [4] => N/A / Description
    [5] => N/A / Opérateur logique
    [8] => N/A / Actif
    [16] => N/A / Commentaires
    [19] => N/A / Dernière modification
    [80] => N/A / Entité
    [86] => N/A / Sous-entités
    [121] => N/A / Date de création
)
```
```122``` can be used for this new ```searchoption```

File: src/Rule.php [hunk 1]
```diff
@@ -790,6 +790,14 @@ public function rawSearchOptions()
             'datatype'           => 'text'
         ];
 
+        $tab[] = [
+            'id'                 => '1234', //Don't know how to determine this ID
```
Comment (stonebuzz) [inline src/Rule.php:794]:
```suggestion
            'id'                 => '122',
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

File: src/Rule.php (hunk 1 above)
Comment (trasher) [inline src/Rule.php:794]:
```suggestion
            'id'                 => '122',
```

**Comment by trasher** [comment]
I made the ID change so tests can be run

**Review by cedric-anne — APPROVED** [review]

**Review by stonebuzz — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**